viking/src/tools/check.rs

```diff
@@ -174,11 +174,13 @@ fn get_function_to_check_from_args(args: &[String]) -> Result<String> {
     Ok(maybe_fn_to_check.remove(0))
 }
 
-fn get_version_from_args(args: &[String]) -> Result<Option<&str>> {
+fn get_version_from_args_or_config(args: &[String]) -> Result<Option<&str>> {
     let mut iter = args.iter().filter_map(|s| s.strip_prefix("--version="));
     match (iter.next(), iter.next()) {
         (Some(_), Some(_)) => bail!("expected only one version number ('--version=XXX')"),
-        (None, None) => Ok(None),
+        (None, None) => Ok(
+            repo::CONFIG.get("default_version").map(|s| s.as_str()).unwrap_or_else(|| None)
```

Comment:
`.unwrap_or_else(|| None)` -> `ok()`

Reply:
That would map `None` to `Err(...)`, which is not correct. Instead, it should unwrap to `None` in that case.

The problem is that `.get()` and then `as_str()` seems to return an `Option<Option<&str>>`, so one instance has to be unwrapped there.

Reply:
Oh, so `repo::CONFIG.get("default_version").map(|s| s.as_str())` is an `Option<Option<...>>`. Thought it was a Result.

`.unwrap_or_else(|| None)` still looks rather silly though -- `.unwrap_or(None)` would be preferable.

Reply:
Done.